### ISSUE TITLE
last_relevant

### ISSUE THREAD

**Issue by PatWie** [PatWie/tfblueprint#69]
> Until `tf.gather_nd()` will have a gradient implementation, I suggest an alternative implementation for 'last_relevant': 
> max_length = int(output.get_shape()[1])
> relevant = tf.reduce_sum(tf.mul(output, tf.expand_dims(tf.one_hot(length, max_length), -1)), 1)
> I ran it on a small benchmark and it seems to improve the running time, due to (I think) better parallelism. 
> Explanation: we first create a matrix where each row has a single '1' entry corresponding to the value in 'length', and the rest are '0'. We then expend its dim in order to multiply it with 'output'. Now each entry whether got a 0-vector or the actual vector as in 'output'. We then compute the sum in order to eliminate the 0-vectors.

https://danijar.com/variable-sequence-lengths-in-tensorflow/
